src/Symfony/Component/Form/Extension/Core/Type/LanguageType.php

```diff
@@ -42,16 +43,38 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults([
             'choice_loader' => function (Options $options) {
                 $choiceTranslationLocale = $options['choice_translation_locale'];
-
-                return new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale) {
-                    return array_flip(Languages::getNames($choiceTranslationLocale));
+                $choiceSelfTranslation = $options['choice_self_translation'];
+
+                return new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale, $choiceSelfTranslation) {
+                    if (true === $choiceSelfTranslation) {
+                        foreach (Languages::getLanguageCodes() as $languageCode) {
+                            try {
+                                $languagesList[$languageCode] = Languages::getName($languageCode, $languageCode);
+                            } catch (\Exception $e) {
+                            }
+                        }
+                    } else {
+                        $languagesList = Languages::getNames($choiceTranslationLocale);
+                    }
+
+                    return array_flip($languagesList);
                 });
             },
+            'choice_self_translation' => false,
```

Comment:
what about make it default `true` if `choice_translation_locale` is `null` else set it to `false`?
that can be achieved with a lazy default function